client/src/component/FileDownloadButtons.tsx

```diff
@@ -1,116 +1,152 @@
 import * as React from 'react';
 
-import { Button, Callout, Intent, Popover } from '@blueprintjs/core';
+import { Button, Callout, Intent, Popover, Spinner } from '@blueprintjs/core';
 
+import deleteFileForCounty from 'corla/action/dos/deleteFileForCounty';
 import downloadFile from 'corla/action/downloadFile';
 
 interface UploadedFileProps {
+    allowDelete: boolean;
     description: string;
-    file: UploadedFile | undefined | null;
+    /* not sure why the type checker doesn't like this definition after changing
+       UploadedFile to a class(?) */
+    /* file: UploadedFile | undefined | null; */
```

Comment:
The new class `UploadedFile` creates an implicit interface in this module which make it ambiguous with the ambient type declaration.

Reply:
oh you can't have two things with the same name, duh - that is hilarious.

Reply:
TypeScript is smart enough to know the difference between vars and types, though, which is why it worked before! It's just that extra type declaration that implicitly comes with the `class` declaration that caused the issue.